pkg/deploy/lattice/service_synthesizer.go

```diff
@@ -32,49 +32,47 @@ func (s *serviceSynthesizer) Synthesize(ctx context.Context) error {
 	// TODO
 	for _, resService := range resServices {
 
-		glog.V(6).Infof("Synthesize Service/HTTPRoute: %v\n", resService)
+		glog.V(6).Infof("SynthesizeCreation Service/HTTPRoute: %v\n", resService)
 		if resService.Spec.IsDeleted {
-			// handle service delete
-			err := s.serviceManager.Delete(ctx, resService)
-
-			if err == nil {
-				glog.V(6).Infof("service - Synthesizer: finish deleting service %v\n", *resService)
-				s.latticeDataStore.DelLatticeService(resService.Spec.Name, resService.Spec.Namespace)
-
-				// also delete all listeners of this service
-				listeners, err := s.latticeDataStore.GetAllListeners(resService.Spec.Name, resService.Spec.Namespace)
-
-				glog.V(6).Infof("service synthesize -- need to delete listeners %v, err : %v\n", listeners, err)
```

Comment:
please keep this debug log, which can be helpful

Reply:
These code moved to here: `func (s *serviceSynthesizer) SynthesizeDeletion(ctx context.Context)`, 


And I didn't add this code
```
listeners, err := s.latticeDataStore.GetAllListeners(resService.Spec.Name, resService.Spec.Namespace)
for _, l := range listeners {
					s.latticeDataStore.DelListener(resService.Spec.Name, resService.Spec.Namespace,
						l.Key.Port, l.Key.Protocol)
				}
```
Because I think, when we delete the latticeService, we use this order:

- listenerSynthesizer.Synthesize(ctx)
- ruleSynthesizer.Synthesize(ctx)
- serviceSynthesizer.SynthesizeDeletion(ctx)

So that we don't need to clean Listeners DataStore in serviceSynthesizer.SynthesizeDeletion(), is that correct?